WebRoot/js/zimbraMail/calendar/view/ZmCalendarPrefsPage.js

```diff
@@ -819,43 +768,22 @@ function(templateId) {
     radioGroup.addSelectionListener(new AjxListener(this, this._toggleNormalCustom));
     this._radioGroup = radioGroup;
 
-    customBtn = new DwtButton({parent:this, parentElement:(this._htmlElId + "_CAL_CUSTOM_WORK_HOURS")});
-    customBtn.setText(ZmMsg.calendarCustomBtnTitle);
-    customBtn.addSelectionListener(new AjxListener(this, this._openCustomizeDlg));
-    customBtn.setEnabled(isCustom);
-    this._customBtn = customBtn;
-};
-
-/**
- * Create the custom work hours dialog box
- * @param parent
- * @param templateId
- * @param workHours the work hours parsed array
- */
-ZmCustomWorkHoursDlg = function (parent, templateId, workHours) {
-    DwtDialog.call(this, {parent:parent});
-    this._workHours = workHours;
-    this._startTimeSelect = [];
-    this._endTimeSelect = [];
-    this._workDaysCheckBox = [];
-    var contentHtml = AjxTemplate.expand("prefs.Pages#"+templateId, {id:this._htmlElId});
-    this.setContent(contentHtml);
-	this.setTitle(ZmMsg.calendarCustomDlgTitle);
-    this._daysChanged = false;
+    //initialize custom work-hours
+    this._initCustomWorkHours();
 };
 
-ZmCustomWorkHoursDlg.prototype = new DwtDialog;
-ZmCustomWorkHoursDlg.prototype.constructor = ZmCustomWorkHoursDlg;
-
-ZmCustomWorkHoursDlg.prototype.initialize = function(workHours) {
+ZmWorkHours.prototype._initCustomWorkHours = function() {
     var i,
         el,
-        checkbox,
+        label,
         startTimeSelect,
         endTimeSelect,
-        inputTime;
+        inputTime,
+        workHours,
+        isCustom;
 
-    workHours = workHours || this._workHours;
+    workHours = this._workHours,
```

Comment:
Ohh, actually issue is of missing semi-colon after `workHours` assignment. Rest looks good to me. 
Let me know if I am missing something.
@tarangkhandelwal 

Thanks